### PR TITLE
Add tint color to Alert api

### PR DESCRIPTION
**Pull request by Ajackster**
This PR adds `tintColor` to the Alert api (**only on iOS**). The `tintColor` parameter is used to change the color of the action button's text. This PR addresses the suggestion from issue #7062

Here are a couple examples.
```
Alert.alert(
    'Invalid credentials',
    '',
    [
      { text: 'OK' }
    ],
    null,
    'green'
);
```
screen shot 2017-02-02 at 5 05 05 pm

```
Alert.alert(
    'Invalid credentials',
    '',
    [
      { text: 'OK' }
    ],
    null,
    'orange'
);
```
screen shot 2017-02-02 at 5 04 32 pm

```
Alert.alert(
      "Are you sure?",
      "Deleting this note will remove it from the server and device.",
      [
        { text: "Delete", onPress: onDeletePress, style: 'destructive' },
        { text: "Cancel" }
      ],
     null,
     'black'
)
```
screen shot 2017-02-02 at 4 59 27 pm

As you can see from the image with the _Delete_ and _Cancel_ button, the style property for the button overrides the `tintColor`.

Would you guys be interested in something like this? OR if there's a better way you would like to see it implemented? Thanks!

### CODE REVIEW

**Comment by henrikra** [comment]
I think you should refactor Alert's all parameters to an object because 4 parameters is way too much :O

**Comment by Ajackster** [comment]
@henrikra That's exactly what I was thinking! I think we'd have to make a different function and add a deprecation warning before we change the `alert` function so people's projects don't break.

Something like...

```
Alert.alertWithProperties({
    title: 'Invalid credentials',
    buttons: [
      { text: 'OK' }
    ],
    tintColor: 'green'
});
```

screen shot 2017-02-02 at 5 05 05 pm

File: Libraries/Alert/AlertIOS.js [hunk 1]
```diff
@@ -65,15 +66,15 @@ type ButtonsArray = Array<{
...
   /**
    * Callback function when button pressed
    */
-  onPress?: ?Function,
+    onPress?: ?Function,
   /**
    * Button style
    */
-  style?: AlertButtonStyle,
+    style?: AlertButtonStyle,
```
Comment (mkonicek) [inline Libraries/Alert/AlertIOS.js]: Can you revert the unrelated changes to spacing in this file please?

File: Libraries/Alert/AlertIOS.js [hunk 2]
```diff
@@ -136,14 +139,15 @@ class AlertIOS {
     title: ?string,
     message?: ?string,
     callbackOrButtons?: ?(() => void) | ButtonsArray,
-    type?: AlertType,
+    tintColor?: ?string,
+    type?: AlertType
```
Comment (mkonicek) [inline Libraries/Alert/AlertIOS.js]: This is a breaking change. Where people passed a `type` in their apps, this will be now treated as the `tintColor` and fail.

**Comment by mkonicek** [comment]
Good idea, thanks! However the code here is a breaking change. Where people passed a type in their apps, this will be now treated as the tintColor and fail. We should to avoid breaking changes whenever possible, otherwise we make it very difficult for people to upgrade to new versions of RN.

I agree the `Alert.alert(` is pretty ugly, and introducing one that accepts an object would be great. No need to introduce a new function with a different name - it is possible to do this in a backwards-compatible way and keep just one function. Check out how it's done in [`ScrollView.scrollTo`](https://github.com/facebook/react-native/blob/master/Libraries/Components/ScrollView/ScrollView.js#L411). Would you be up for doing that for the `Alert` API so it's nice once and for all? :)

**Comment by Ajackster** [comment]
Sure thing, I'd love to work on it!

**Comment by ericvicenti** [comment]
Can you restructure the code to have less duplication? You can simply assign title correctly before doing platform-specific behavior.

Also, can you please update the documentation to describe the new object API?

The flow type looks good but I think it should be broken out into a separate, exported type definition. You can search the codebase for `export type` to see examples of this.

**Comment by hramos** [comment]
Are you still working on this PR?

**Comment by hramos** [comment]
Closing due to lack of response. If you're still working on this, just let us know and we can reopen!
